app/javascript/availability/components/location_info.jsx

```diff
@@ -0,0 +1,95 @@
+          role="status" aria-hidden="true"></span>
+);
+
+const LocationInfo = (props) => {
+    const holding = props.holding;
+    
+    // Location information presented to the user is different based on a few scenarios
+    // First, if it's related to ILL
+    if (availability.isIllLink(holding)) {
+        return (
+            <a 
+                data-type="ill-link" 
+                data-catkey={holding.catkey}
+                data-call-number={holding.callNumber} 
+                data-link-type={availability.illLinkType(holding)}
+                data-item-location={holding.locationID}
+                href="#"
+            >
+                <Spinner />
+                {availability.illLinkText(holding)}
+            </a>
+        )
+    }
+
+    // AEON
+    if (availability.isArchivalThesis(holding)) {
+        const illiadURL = "https://psu.illiad.oclc.org/illiad/upm/lending/lendinglogon.html";
+        const aeonLocationText = availability.mapLocation(availability.allLocations, holding.locationID);
+
+        return (<>
+            <a 
+                data-type="ill-link" 
+                data-catkey={holding.catkey}
+                data-call-number={holding.callNumber}
+                data-link-type="archival-thesis" 
+                data-item-type={holding.itemTypeID}
+                href="#">
+                    <Spinner />
+                    Request Scan - Penn State Users
+            </a><br />
+
+            <a href={illiadURL}>Request Scan - Guest</a><br />
+
+            <a 
+                data-type="aeon-link" 
+                data-catkey={holding.catkey}
+                data-call-number={holding.callNumber}
+                data-link-type="archival-thesis"
+                data-item-type={holding.itemTypeID}
+                data-item-id={holding.itemID}
+                data-item-location={aeonLocationText}
+                href="#">
+                    <Spinner />
+                    View in Special Collections
+            </a>
+        </>)
+    }
+
+    // Special Collections
+    if (availability.isArchivalMaterial(holding)) {
+        const specialCollectionsText = availability.mapLocation(availability.allLocations, holding.locationID);
+
+        return (<>
+            {specialCollectionsText}<br /> 
+
+            <a 
+                data-type="aeon-link" 
+                data-catkey={holding.catkey}
+                data-call-number={holding.callNumber}
+                data-link-type="archival-material"
+                data-item-type={holding.itemTypeID}
+                data-item-id={holding.itemID}
+                data-item-location={specialCollectionsText}
+                href="#">
+                    <Spinner />
+                    Request Material
+            </a>
+        </>);
+    }
+
+    // Otherwise use the default translation map for location display, no link
+    return availability.mapLocation(availability.allLocations, holding.locationID);
```

Comment:
Your second argument is the same as "self", i.e. `availability`. Wouldn't the `mapLocation` function be able to call `allLocations` from within itself?

Reply:
Ha! Yeah, this was just a copy paste, and it seems the first argument to `mapLocation` is always `allLocations`. Will change that. Good catch!

Reply:
@awead - moved the `mapLocation` function into this component and also removed the unnecessary argument. Whittling away at the mega file little by little. 🙂 

If you and @banukutlu are ok with it, I'd like to handle the AEON/ILL stuff in a separate PR, since this one is getting kinda big already.